source/Octopus.Client/Repositories/RunbookRepository.cs

```diff
@@ -45,9 +48,40 @@ public RunbookRunPreviewResource GetPreview(DeploymentPromotionTarget promotionT
             return Client.Get<RunbookRunPreviewResource>(promotionTarget.Link("RunbookRunPreview"));
         }
 
-        public RunbookRunResource Run(RunbookResource runbook, RunbookRunResource runbookRun)
+        public RunbookRunResource RunPublished(string projectName, string runbookName, string environmentName, string tenantName, Dictionary<string, string> formValues = null)
+        {
+            return RunPublishedInner(projectName, runbookName, environmentName, tenantName, formValues);
+        }
+
+        public RunbookRunResource RunPublished(string projectName, string runbookName, string environmentName, Dictionary<string, string> formValues = null)
+        {
+            return RunPublishedInner(projectName, runbookName, environmentName, null, formValues);
+        }
+
+        private RunbookRunResource RunPublishedInner(string projectName, string runbookName, string environmentName, string tenantName, Dictionary<string, string> formValues)
+        {
+            var root = Repository.LoadRootDocument();
+
+            RunPublishedRunbookResource runPublished = new RunPublishedRunbookResource()
+            {
+                ProjectName = projectName,
+                RunbookName = runbookName,
+                EnvironmentName = environmentName,
+                TenantName = tenantName
+            };
+
+            if (formValues != null)
+                runPublished.FormValues = formValues;
+            
+            return Client.Post<object, RunbookRunResource>(root.LinkToRunbooksRunPublished(), runPublished);
+        }
+    }
+    
+    internal static class RunbookLinkExtensions
```

Comment:
Worth extracting this?

Reply:
Up to you. I think this is nicely discoverable here tbh